docs/source/guides/inference.mdx

```diff
@@ -1,58 +1,314 @@
+service that runs inference for hosted models. There are several services you can connect to:
+- [Inference API](https://huggingface.co/docs/api-inference/index): a service that allows you to run accelerated inference
+on Hugging Face's infrastructure and for free. This service is a fast way to get started, tests different models and
+prototype AI products.
+- [Inference Endpoints](https://huggingface.co/inference-endpoints): a product to easily deploy models to production.
+Inference is run by Hugging Face in a dedicated, fully managed infrastructure on the Cloud of your choice.
+- [SageMaker Hugging Face Inference Toolkit](https://github.com/aws/sagemaker-huggingface-inference-toolkit)
+
+All of these services can be called with the [`InferenceClient`] object. Let's see how to use it!
+
+<Tip>
+
+[`InferenceClient`] is a Python client making HTTP calls to our APIs. If you want to make the HTTP calls directly, please
+refer to the [Inference API](https://huggingface.co/docs/api-inference/index) or to the
+[Inference Endpoints](https://huggingface.co/docs/inference-endpoints/index) documentation pages.
+
+For web development, a [JS client](https://huggingface.co/docs/huggingface.js/inference/README) has been released.
+If you are interested in game development, you might have a look to our [C# project](https://github.com/huggingface/unity-api).
+
+</Tip>
 
 <Tip>
 
-If you want to make the HTTP calls directly, please refer to [Accelerated Inference API Documentation](https://api-inference.huggingface.co/docs/python/html/index.html) or to the sample snippets visible on every supported model page.
+[`InferenceClient`] acts as a replacement for the legacy [`InferenceApi`] client. It adds specific support for tasks and
+handles inference on both [Inference API](https://huggingface.co/docs/api-inference/index) and [Inference Endpoints](https://huggingface.co/docs/inference-endpoints/index).
+Learn how to migrate to the new client in this section.
 
 </Tip>
 
-<div class="flex justify-center">
-<img class="block dark:hidden" src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/hub/inference_api_snippet.png"/>
-<img class="hidden dark:block" src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/hub/inference_api_snippet-dark.png"/>
-</div>
+## Getting started
 
-Begin by creating an instance of the [`InferenceApi`] with the model repository ID of the model you want to use. You can find your `API_TOKEN` under Settings from your Hugging Face account. The `API_TOKEN` will allow you to send requests to the Inference API.
+Let's get started with a first example:
 
 ```python
->>> from huggingface_hub.inference_api import InferenceApi
->>> inference = InferenceApi(repo_id="bert-base-uncased", token=API_TOKEN)
+>>> from huggingface_hub import InferenceClient
+>>> client = InferenceClient()
```

Comment:
I would rather not add a token parameter. I feel that if we add it everywhere we encourage users to pass tokens as argument instead of log in correctly.